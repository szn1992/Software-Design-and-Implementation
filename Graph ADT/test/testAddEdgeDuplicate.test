# Test adding duplicate edges, the label between a parent node
# and a child node should always be unique

# Create a graph
CreateGraph graph1

# Add 3 nodes
AddNode graph1 n1
AddNode graph1 n2
AddNode graph1 n3

# Add 6 edges, last 3 are duplicates
AddEdge graph1 n1 n1 e1
AddEdge graph1 n1 n2 e2
AddEdge graph1 n1 n3 e3
AddEdge graph1 n1 n1 e1
AddEdge graph1 n1 n2 e2
AddEdge graph1 n1 n3 e3

# Print the nodes
ListNodes graph1

# Print the outgoing edges from n1
ListChildren graph1 n1