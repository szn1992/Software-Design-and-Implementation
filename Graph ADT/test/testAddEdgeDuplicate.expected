# Test adding duplicate edges, the label between a parent node
# and a child node should always be unique

# Create a graph
created graph graph1

# Add 3 nodes
added node n1 to graph1
added node n2 to graph1
added node n3 to graph1

# Add 6 edges, last 3 are duplicates
added edge e1 from n1 to n1 in graph1
added edge e2 from n1 to n2 in graph1
added edge e3 from n1 to n3 in graph1
added edge e1 from n1 to n1 in graph1
added edge e2 from n1 to n2 in graph1
added edge e3 from n1 to n3 in graph1

# Print the nodes
graph1 contains: n1 n2 n3

# Print the outgoing edges from n1
the children of n1 in graph1 are: n1(e1) n2(e2) n3(e3)